### PR TITLE
document different ways of setting up NC

### PR DESCRIPTION
**Pull request by individual-it**
document various general ways to configure and setup NC that are useful for testing

fixes https://community.openproject.org/projects/nextcloud-integration/work_packages/details/43529

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -152,6 +152,51 @@ docker compose exec --user www-data nextcloud php occ config:system:set allow_lo
 - configure the connection to OpenProject using `http://host.docker.internal:3000` as the OpenProject URL
 - in OpenProject use `http://localhost:8080` as the NextCloud URL
 
+#### change the setup for testing purpose
```
Comment (kiranparajuli589) [inline README.md:155]:
```suggestion
#### Change the setup for testing purpose
```
just to maintain consistency

File: README.md [hunk 2]
```diff
@@ -152,6 +152,51 @@ docker compose exec --user www-data nextcloud php occ config:system:set allow_lo
...
+##### strip off Authorization Bearer tokens of HTTP requests
```
Comment (kiranparajuli589) [inline README.md:156]:
```suggestion
##### Strip off Authorization Bearer tokens of HTTP requests
```

File: README.md [hunk 3]
```diff
@@ -152,6 +152,51 @@ docker compose exec --user www-data nextcloud php occ config:system:set allow_lo
...
+docker compose exec nextcloud a2dismod rewrite
+docker compose exec nextcloud service apache2 restart
+```
+To enable it again run:
+```shell
+docker compose exec nextcloud a2enmod rewrite
+docker compose exec nextcloud service apache2 restart
+```
+
+##### change version of Nextcloud
```
Comment (kiranparajuli589) [inline README.md:169]:
```suggestion
##### Change version of Nextcloud
```

File: README.md [hunk 4]
```diff
@@ -152,6 +152,51 @@ docker compose exec --user www-data nextcloud php occ config:system:set allow_lo
...
+
+  cron:
+    image: nextcloud:23-apache
+```
+
+Please note:
+1. only [apache based versions](https://hub.docker.com/_/nextcloud/?tab=description) will work
+2. Nextcloud does not support downgrading, so if you want to go back to an older version, you need to delete all the volumes with `docker compose down -v` and start the Nextcloud installation again
+
+##### disable pretty URLs
```
Comment (kiranparajuli589) [inline README.md:187]:
```suggestion
##### Disable pretty URLs
```

File: README.md [hunk 5]
```diff
@@ -152,6 +152,51 @@ docker compose exec --user www-data nextcloud php occ config:system:set allow_lo
...
+    image: nextcloud:23-apache
+    ports:
+      - "8080:80"
+
+  cron:
+    image: nextcloud:23-apache
+```
+
+Please note:
+1. only [apache based versions](https://hub.docker.com/_/nextcloud/?tab=description) will work
```
Comment (kiranparajuli589) [inline README.md:184]:
```suggestion
1. Only [apache based versions](https://hub.docker.com/_/nextcloud/?tab=description) will work
```

File: README.md [hunk 6]
```diff
@@ -152,6 +152,51 @@ docker compose exec --user www-data nextcloud php occ config:system:set allow_lo
...
+2. Nextcloud does not support downgrading, so if you want to go back to an older version, you need to delete all the volumes with `docker compose down -v` and start the Nextcloud installation again
+
+##### disable pretty URLs
+NC can work with or without `index.php` in the URL. The connection to Openproject has to work regardless on how Nextcloud is configured.
```
Comment (kiranparajuli589) [inline README.md:188]:
```suggestion
Nextcloud can work with or without `index.php` in the URL. The connection to Openproject has to work regardless of how it is configured.
```

File: README.md [hunk 7]
```diff
@@ -152,6 +152,51 @@ docker compose exec --user www-data nextcloud php occ config:system:set allow_lo
...
+
+By default, the docker setup in this repo starts Nextcloud without `index.php` in the URL, to change that we have to edit the `.htaccess` file:
```
Comment (kiranparajuli589) [inline README.md:190]:
```suggestion
By default, the docker setup in this repo starts Nextcloud without `index.php` in the URL. To change that, we have to edit the `.htaccess` file:
```

File: README.md [hunk 8]
```diff
@@ -152,6 +152,51 @@ docker compose exec --user www-data nextcloud php occ config:system:set allow_lo
...
+    ports:
+      - "8080:80"
+
+  cron:
+    image: nextcloud:23-apache
+```
+
+Please note:
+1. only [apache based versions](https://hub.docker.com/_/nextcloud/?tab=description) will work
+2. Nextcloud does not support downgrading, so if you want to go back to an older version, you need to delete all the volumes with `docker compose down -v` and start the Nextcloud installation again
```
Comment (kiranparajuli589) [inline README.md:185]:
```suggestion
2. Nextcloud does not support downgrading. If you want to go back to an older version, you need to delete all the volumes with `docker compose down -v` and start the Nextcloud installation again
```

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1: some english/grammar nitpicks.

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

File: README.md [hunk 9]
```diff
@@ -152,6 +152,51 @@ docker compose exec --user www-data nextcloud php occ config:system:set allow_lo
...
+Please note:
+1. Only [apache based versions](https://hub.docker.com/_/nextcloud/?tab=description) will work
+2. Nextcloud does not support downgrading. If you want to go back to an older version, you need to delete all the volumes with `docker compose down -v` and start the Nextcloud installation again
+
+##### Disable pretty URLs
+Nextcloud can work with or without `index.php` in the URL. The connection to Openproject has to work regardless of how it is configured.
+
+By default, the docker setup in this repo starts Nextcloud without `index.php` in the URL. To change that, we have to edit the `.htaccess` file:
+```shell
+docker compose exec --user www-data nextcloud sed -i '/#### DO NOT CHANGE ANYTHING ABOVE THIS LINE ####/,$d' .htaccess
```
Comment (wielinde) [inline README.md:192]: @individual-it What does this line actually do? How does it enable `index.php` in the paths? Could you please explain it a bit? What does `sed` command do? I am asking as I really want to understand in what state the installation will be in when I execute that command.

Comment (individual-it) [reply README.md:210]: done in 6690799ede93acf4dd6d67f7b9df3d2e9f9920b4

**Review by Kharonus — APPROVED** [review]
I like it. And with the latest changes, we can basically just check the repo multiple times and change every one to another test case (e.g. having one running a nextcloud 22 without pretty urls and another one with `latest` and pretty urls).

And if there is a need for more differentiated testing, then we can always create more setups easily with that documentation.

**Review by SwikritiT — APPROVED** [review]

File: README.md [hunk 10]
```diff
@@ -152,6 +152,71 @@ docker compose exec --user www-data nextcloud php occ config:system:set allow_lo
...
+docker compose exec nextcloud service apache2 restart
+```
+To enable it again run:
+```shell
+docker compose exec nextcloud a2enmod rewrite
+docker compose exec nextcloud service apache2 restart
+```
+
+##### Change version of Nextcloud
+To test another version of Nextcloud change the nextcloud images in the `docker-compose.overwrite.yml`
```
Comment (kiranparajuli589) [inline README.md:170]:
```suggestion
To test another version of Nextcloud change the nextcloud images in the `docker-compose.override.yml`
```

**Pull request merged**